src/cursor/cur_hs.c

```diff
@@ -645,16 +645,66 @@ __curhs_insert(WT_CURSOR *cursor)
         hs_tombstone = NULL;
     }
 
+retry:
```

Comment:
I believe this is a bug fix in unrelated code but I am indifferent if you fix it here or open a new ticket.